Link to the data set:
https://www.kaggle.com/datasets/swaptr/layoffs-2022?resource=download

In [7]:
import pandas as pd

df = pd.read_csv("Data/layoffs.csv")
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 4600 entries, 0 to 4599
Data columns (total 11 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   company              4600 non-null   str    
 1   location             4599 non-null   str    
 2   total_laid_off       3004 non-null   float64
 3   date                 4600 non-null   str    
 4   percentage_laid_off  2883 non-null   float64
 5   industry             4598 non-null   str    
 6   source               4597 non-null   str    
 7   stage                4592 non-null   str    
 8   funds_raised         4054 non-null   float64
 9   country              4598 non-null   str    
 10  date_added           4600 non-null   str    
dtypes: float64(3), str(8)
memory usage: 395.4 KB


In [8]:
df.head()

,company,location,total_laid_off,date,percentage_laid_off,industry,source,stage,funds_raised,country,date_added
0,Neo Financial,"Calgary, Non-U.S.",102.0,9/8/2026,0.10,Finance,https://www.linkedin.com/pulse/important-chang...,Series D,544.0,Canada,9/9/2026
1,Samsung,"Gurugram, Non-U.S.",80.0,9/8/2026,NaN,Hardware,https://timesofindia.indiatimes.com/business/i...,Post-IPO,NaN,India,9/9/2026
2,Harrison.ai,"Melbourne, Non-U.S.",NaN,9/7/2026,NaN,Healthcare,https://www.abc.net.au/news/2026-09-08/harriso...,Series C,253.0,Australia,9/9/2026
3,PayPal,"Tel Aviv, Non-U.S.",70.0,9/6/2026,NaN,Finance,https://www.jpost.com/business-and-innovation/...,Post-IPO,216.0,Israel,9/9/2026
4,The Trade Desk,Los Angeles,585.0,9/3/2026,0.15,Marketing,https://www.adweek.com/programmatic/the-trade-...,Post-IPO,257.0,United States,9/4/2026


## Data Dictionary

The dataset has 4,600 rows and 11 columns. Each row is one layoff event at a company.

| Column | Dtype | Non-Null | Use |
|---|---|---|---|
| company | str | 4600 | Which company did the layoff. Identifies each event. |
| location | str | 4599 | HQ city, tagged "Non-U.S." for international companies. Geographic comparisons. |
| total_laid_off | float64 | 3004 | Headcount cut in the event — the main size measure. Missing in ~35% of rows because news reports don't always give a number. |
| date | str | 4600 | Date of the layoff. Needs pd.to_datetime conversion for time-based analysis. |
| percentage_laid_off | float64 | 2883 | Share of the workforce cut (0 to 1). A value of 1.0 means the company shut down entirely. Also missing in many rows. |
| industry | str | 4598 | Company's sector (Finance, Crypto, Healthcare, etc.). Main categorical grouping. |
| source | str | 4597 | URL of the news article reporting the layoff. Shows data provenance; not used in analysis. |
| stage | str | 4592 | Funding stage at the time (Seed, Series A–C, Post-IPO). Proxy for company maturity. |
| funds_raised | float64 | 4054 | Total funding the company raised, in $ millions. |
| country | str | 4598 | Country of the company. Geographic grouping. |
| date_added | str | 4600 | When the row was added to the tracker (vs. date = when the layoff happened). Bookkeeping only. |

**Questions**
1. How have layoffs trended since 2020, was the 2022–23 wave bigger than the 2020 COVID wave? Does 2026 look like a third wave or a cooldown?
2. Which funding stages produce the most complete shutdowns, and did shutdowns spike in any particular year?
3. Which industries got hit hardest overall, and did the mix change over time? (Crypto in 2022 vs. AI-adjacent sectors now)?